contracts/extensions/MultisigPermissions.sol

```diff
@@ -0,0 +1,673 @@
+    require(
+      colony.hasUserRole(msgSender(), 1, ColonyDataTypes.ColonyRole.Root),
+      "multisig-permissions-not-core-root"
+    );
+    _;
+  }
+
+  function setGlobalThreshold(uint256 _globalThreshold) public onlyCoreRoot {
+    globalThreshold = _globalThreshold;
+
+    emit GlobalThresholdSet(_globalThreshold);
+  }
+
+  function setDomainSkillThreshold(uint256 _domainSkillId, uint256 _threshold) public onlyCoreRoot {
+    domainSkillThreshold[_domainSkillId] = _threshold;
+
+    emit DomainSkillThresholdSet(_domainSkillId, _threshold);
+  }
+
+  function createMotion(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    address[] memory _targets,
+    bytes[] memory _data
+  ) public notDeprecated {
+    require(_targets.length == _data.length, "colony-multisig-invalid-motion");
+    require(_targets.length >= 1, "colony-multisig-invalid-motion");
+
+    motionCount += 1;
+    Motion storage motion = motions[motionCount];
+    motion.targets = _targets;
+    motion.data = _data;
+    motion.creator = msgSender();
+    motion.creationTimestamp = block.timestamp;
+
+    for (uint256 i = 0; i < motion.data.length; i += 1) {
+      uint256 actionDomainSkillId;
+      bytes32 actionRequiredPermissions;
+
+      (actionDomainSkillId, actionRequiredPermissions) = getActionSummary(
+        motion.data[i],
+        _targets[i]
+      );
+
+      // slither-disable-next-line incorrect-equality
+      if (motion.domainSkillId == 0 && motion.requiredPermissions == 0) {
+        motion.domainSkillId = actionDomainSkillId;
+        motion.requiredPermissions = actionRequiredPermissions;
+      } else {
+        // slither-disable-next-line incorrect-equality
+        require(
+          motion.domainSkillId == actionDomainSkillId &&
+            motion.requiredPermissions == actionRequiredPermissions,
+          "colony-multisig-invalid-motion"
+        );
+      }
+    }
+
+    validateMotionDomain((_permissionDomainId), _childSkillIndex, motionCount);
+    validateUserPermissions(_permissionDomainId, motionCount);
+
+    emit MotionCreated(msgSender(), motionCount);
+
+    changeVoteFunctionality(
+      _permissionDomainId,
+      _childSkillIndex,
+      motionCount,
+      Vote.Approval,
+      true
+    );
+  }
+
+  function changeVote(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _motionId,
+    Vote _vote
+  ) public motionExists(_motionId) notExecuted(_motionId) notRejected(_motionId) {
+    changeVoteFunctionality(
+      _permissionDomainId,
+      _childSkillIndex,
+      _motionId,
+      Vote.Approval,
+      _vote == Vote.Approval
+    );
+
+    changeVoteFunctionality(
+      _permissionDomainId,
+      _childSkillIndex,
+      _motionId,
+      Vote.Rejection,
+      _vote == Vote.Rejection
+    );
+  }
+
+  function changeVoteFunctionality(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _motionId,
+    Vote _vote,
+    bool _setTo
```

Comment:
Is there any way to dispense with this argument? What information does it provide above-and-beyond `_vote`? I see it's used to call `changeVoteFunctionality` twice, but I can't imagine this is the best implementation.

Reply:
Maybe I'm not understanding what you're looking for here. Your top-level suggestion was 

> A single changeVote function should suffice, and would allow you to avoid passing in two variables (_type and _setTo)

In order to provide functionality where users can remove approvals or rejections, the implication of that suggestion (I believe) is that if I call `changeVote(REJECTION)`, it touches the storage for both rejections (to record that I am now voting for a rejection) and approvals (to record that, if I was voting for approval, I am no longer).

That common functionality is factored out in to `changeVoteFunctionality`. It could all be one function, but there would be a lot of duplication, which I assume you're against as well.